components/SideBar.vue

```diff
@@ -51,38 +57,47 @@
 
 <script lang="ts" setup>
 import { CIcon } from "@coreui/icons-vue";
+import type { VersionData } from "@/types/apiResponseTypes";
+
+const { data: versionData } = useFetch("/api/versions") as { data: Ref<VersionData> };
```

Comment:
`ServerApiResponse` was named to try and distinguish it from the R API, so that'll have to be re-named... if we simply call it `ApiResponse` might that actually be clearer?

Reply:
Though `ServerApiResponse`'s purpose differs from what you're talking about - `ServerApiResponse` (well, types extended from it) are to be used inside the API routes to dictate the format of the response's data, rather than to be used wrapping a useFetch.

Reply:
Yeah, exactly. I think probably the response types from the R API (used in the web app backend) and the types from the web app backend (used in the front end) should probably live in different modules, and maybe a "shared" module for ones that happen to be the same, with whatever generic classes are useful in the two contexts. Maybe `ServerApiResponse` and `RApiResponse` (I think `DaedalusApiResponse` would be more correct, but very verbose, and you already have rApi module...)